modules/uhttp/client/client_filters_test.go

```diff
@@ -67,6 +74,47 @@ func TestExecutionChainFiltersError(t *testing.T) {
 	assert.Nil(t, resp)
 }
 
+func withOpentracingSetup(t *testing.T, fn func(tracer opentracing.Tracer)) {
+	tracer, closer, err := tracing.InitGlobalTracer(&jconfig.Configuration{}, "Test", ulog.NoopLogger, tally.NullStatsReporter)
+	defer closer.Close()
+	assert.NotNil(t, closer)
+	require.NoError(t, err)
+
+	_serviceName = "test_service"
+	auth.UnregisterClient()
+	defer auth.UnregisterClient()
+	auth.SetupClient(nil)
+	defer auth.SetupClient(nil)
+	fn(tracer)
+}
+
+func TestExecutionChainFilters_AuthContextPropagation(t *testing.T) {
+	withOpentracingSetup(t, func(tracer opentracing.Tracer) {
+		execChain := newExecutionChain(
+			[]Filter{FilterFunc(authenticationFilter)}, getContextPropogationClient(t),
+		)
+		span := tracer.StartSpan("test_method")
+		span.SetBaggageItem(auth.ServiceAuth, _serviceName)
+		ctx := &fxcontext.Context{
+			Context: opentracing.ContextWithSpan(context.Background(), span),
+		}
+		resp, err := execChain.Do(ctx, _req)
+		assert.NoError(t, err)
+		assert.Equal(t, _respOK, resp)
+	})
+}
+
+func getContextPropogationClient(t *testing.T) BasicClient {
+	return BasicClientFunc(
+		func(ctx fx.Context, req *http.Request) (resp *http.Response, err error) {
+			span := opentracing.SpanFromContext(ctx)
+			assert.NotNil(t, span)
```

Comment:
This assert is just making sure that span is never nil. In case if we use getContextPropogationClient in other tests. second one is just verifying that baggage is present.